### PR TITLE
Refactorings of getAminoAcidMapping

### LINKED ISSUE #186: Refactor AminoAcidMapping.getAminoAcidMapping(String)
I've selected [**AminoAcidMapping.getAminoAcidMapping(String)**](https://github.com/Verisimilitude11/DNAnalyzer/blob/43151e8f70d86b468dc676c3f5e3f1bc7a4bdca3/src/main/java/DNAnalyzer/AminoAcidMapping.java#L26-L91) for refactoring, which is a unit of **65** lines of code and **20** branch points. Addressing this will make our codebase more maintainable and improve [Better Code Hub](https://bettercodehub.com)'s **Write Simple Units of Code** guideline rating! 👍 

Here's the gist of this guideline:
- **Definition** 📖 
Limit the number of branch points (if, for, while, etc.) per unit to 4.
- **Why**❓ 
Keeping the number of branch points low makes units easier to modify and test.
- **How** 🔧 
Split complex units with a high number of branch points into smaller and simpler ones.

You can find more info about this guideline in [Building Maintainable Software](http://shop.oreilly.com/product/0636920049159.do). 📖 
 
---- 
ℹ️ To know how many _other_ refactoring candidates need addressing to get a guideline compliant, select some by clicking on the 🔲  next to them. The risk profile below the candidates signals (✅) when it's enough! 🏁 

----
Good luck and happy coding! :shipit: :sparkles: :100:

### PR DESCRIPTION
**Pull request by Speedro**
closes #186
- updated junit version so we can use parametrized tests
- created CodonDataUtilsTest class
- deleted no longer used AminoAcidMapping.java and AminoAcidNames.java.
- only AminoAcid.java enum should be used

### CODE REVIEW

**Comment by Verisimilitude11** [comment]
@Speedro looks like there are conflicts again. This PR is first in my backlog, so I'll review it once you clear the merge conflicts. Thanks!

**Review by Verisimilitude11 — CHANGES_REQUESTED** [review]
@Speedro looks like there are conflicts again. This PR is first in my backlog, so I'll review it once you clear the merge conflicts. Thanks!

File: src/main/java/DNAnalyzer/aminoAcid/AminoAcid.java [hunk 1]
```diff
@@ -36,7 +36,8 @@ public enum AminoAcid {
   SERINE("Serine", "s", "serine", "ser"),
   THREONINE("Theornine", "t", "threonine", "thr"),
   VALINE("Valine", "v", "valine", "val"),
-  THRYPTOPHAN("Thryptophan", "w", "tryptophan", "trp");
+  TRYPTOPHAN("Thryptophan", "w", "tryptophan", "trp"),
```
[the remark this answers is not in the document]
Comment (Speedro) [reply src/main/java/DNAnalyzer/aminoAcid/AminoAcid.java]: @Verisimilitude11 thanks, I resolved the conflicts and left the 'Tryptophan' inside the enum. Please check, if it's ok and let me know if not 🙂 🚀

File: src/test/java/DNAnalyzer/CodonDataUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+import DNAnalyzer.codon.CodonDataUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * CodonDataUtilsTest contains 20 unit tests, one for each acid.
+ *
+ * @author @Speedro
+ */
+public class CodonDataUtilsTest {
```
Comment (Speedro) [inline src/test/java/DNAnalyzer/CodonDataUtilsTest.java:28]: @Verisimilitude11 Ok, I am not sure about the CodeSceneDelta analyzis fail as the strings are valid inputs for the test case. Maybe this analyzis could be softened for test classes or something.

Comment (Speedro) [reply src/test/java/DNAnalyzer/CodonDataUtilsTest.java:28]: Or the dna parts (ATT, ATC etc.) could be turned into String constants. Not sure it that helps though, but maybe could.

Comment (Verisimilitude11) [reply src/test/java/DNAnalyzer/CodonDataUtilsTest.java:28]: Yeah don't worry about CodeScene, they don't have the most reliable reporting. I'll merge later tomorrow after some testing.

**Review by Verisimilitude11 — APPROVED** [review]

**Pull request merged**